### ISSUE TITLE
UnboundLocalError: local variable 'value' referenced before assignment

### ISSUE THREAD

**Issue by steven2308** [wbond/asn1crypto#153]
**Describe the bug**
For some certificates which I assume are wrongly built, the code reaches a path were it tries to use a a variable which has not been defined, especifcally when accessing IPAddress.native

**To Reproduce**
```
import asn1crypto.x509
x509 = b'0\x82\x04\xf00\x82\x02\xd8\xa0\x03\x02\x01\x02\x02\x10\x1b6\n-\x13\xc5>]I\xed\xb3\xce\xd1\xcb\x12~0\r\x06\t*\x86H\x86\xf7\r\x01\x01\x0b\x05\x000\x121\x100\x0e\x06\x03U\x04\n\x13\x07Acme Co0\x1e\x17\r190820155018Z\x17\r200819155018Z0\x121\x100\x0e\x06\x03U\x04\n\x13\x07Acme Co0\x82\x02"0\r\x06\t*\x86H\x86\xf7\r\x01\x01\x01\x05\x00\x03\x82\x02\x0f\x000\x82\x02\n\x02\x82\x02\x01\x00\x9c\x1b\xe2\xe5\xe8\x9f\x18\xe2\xf7\x13!\xaa\xa8\teS\xbb\xbb\x0bo\xab\x8a\xc2\xf5\xe9\xcd\x83NW&\x93\xe3\xc1\xe6\xa9 oP\x1b\xf9%&\xf1\x92Z\x93\xd03\x93\xbe\x0em\x04\xda\xb7\xea\xfa\xd1\'7\x01\xc8<*\xfe\xec}J\x93\xa0|sS~\x8e\xce88\xd4\xf8\x9dHt\x9a\xc7rF\xaf"\xf1\x08\x9e\x11\xa0\x9f\xbco\x16M\xc9\x9c\xfd\xafh".\xe3l0\xf4\xecC\x8e0&>\xc1\xde\t\x85\x0c\x85\xd7\x19\x02\x17\xb4\xdd\x95>\x94P\x81\x10\xda\x0cLk9\x93\xc7u\x03\x1a\xb9\x08D\xa3\xe7\x85\xd8\xa2hS\xf9\xaeCm\x07\xadIG\xffS\x05+M|>\xa8(T\xc6\xb9\xad\xeb\x17\xfc\n\x18\x1d\x90~\x84\xf8a\x8f\xcbJ\xe4\xbb\xd94I\xa7h\x06\x165\xdbF \x8a+\x9eFEm\x06\xcaJ\xe7\xd1\x0b\x8f\xcaY\x0e\xb5#\xbc\x08w@\xb2\xbe-O\xd1\r\xc4\x9a\x96\x1f|\xcdaO\xe4\x8e\xcd\x8d\xcalA\xf6\xad\xe8\xb1k\xb7\x80\xdd\x9d\xc2]\xa2\xda\xdf\\7ep\xb0\x7f2\xc6\x8f\x10K\x06[\x10\xb9c\x83\xc7\xfft$\x8e\xc16\xd8\x820d\x8f\xca\x18!\xd4N\x1bY}\x1fIi\xf8f-\r\xfc\xb6\xcb\xec\x95\xae\xc0\x17\x80\xe6\xcc\xe0\xab\x1fvB,g\xb9\x10\xf9"q\xcam\xb6\xde3^\xbe\xc8o?~\xd1\x8a\x82bZ\xe8R\xb7\x0c\x00\xfa\x1d#9a\xac\xde-\xb9\xb9\xaf\xa7\xf8U-d?V\xb2\xa5\x87\xd2\xec-\x80\xb9&\xf7\x9a\xf8J\x87\xa0Szs\x16\xa582\x10?\xa7\x97\xc5C\x85\x91\x17K\xea\xde%}r\x8a\x19\xed\xaf\x87}M\xe2\xa8t\xbe\'\xf5\xa5&\xa8\x8b\xd7\x85\xe5\xa5\xa8\xde"\x8f\xa5\xc5\x96\x1c\x0b[_\xbcq\x11\xd7q3\xa4\xb4\xef\x0c\xa9G\xebEw\xecXt\xd7)\xf8\x01\xa2e\x07\xc8\xe9\x88\x1a\x14_dv\xf1]~\xdd-\xa6\x14\x19\x8a]:D\xf0\xa9\xffE2\xde/\x15\x05!L\xed\xf6\xa4\xfb*\x8d\x1a{p\x83Y\xe7K\x8e\xed%\xf5\xf3C\x88\xa8\xe3\x02\x03\x01\x00\x01\xa3B0@0\x0e\x06\x03U\x1d\x0f\x01\x01\xff\x04\x04\x03\x02\x05\xa00\x13\x06\x03U\x1d%\x04\x0c0\n\x06\x08+\x06\x01\x05\x05\x07\x03\x010\x0c\x06\x03U\x1d\x13\x01\x01\xff\x04\x020\x000\x0b\x06\x03U\x1d\x11\x04\x040\x02\x87\x000\r\x06\t*\x86H\x86\xf7\r\x01\x01\x0b\x05\x00\x03\x82\x02\x01\x00\x13 \xe0D\xa9\x86\x84qO\xd4o\n\x07I3\xdd\xc8\xd3?\xc8&\x01\xb7\x98\x90\xdf\xfce\x9dm\xcbo\xad8\xd9\x80\xf1\x15\xb4\xb28G\xac%\x81\xdb\xf9\xfe2\t+\xd2\xeb\xbf\xe6M\xcb\'\xc8$\x92\xf1\xe48{\xf1\xad\xee\xf60n\xd5\xa1\x904\x98\x84t\xa5}n4\xfd\xefz\x88`\x9a/\xa2\xa9\x9c\xdb1\xf9\x1aX\x13\x01T@"\xdc\x7f\x93J\xcb\x8b6\x9d\x85y ?n\',\xfd\xd7d\x86%\xcd\xe0\xbdu\x1b\r\xf6u\xfc)\xd5\xaf1S\x12\x11\r\x98\xd3\xb97\x92\x19 \x06PM\xb5\xdeK\x04\x9a\xafx\xf6\\\x12\xdbK\xd9\xdf\x19vc\xcf\x9cI9\x91\xba\xa6\xa2\x1c`z@\xc6\xbcQ\xd3\xc6z\xa0\xf2O\xfe\xa8\x9f\xa2\x1a\xe03\xb3\x1f&+S\x8a\xb3\x98=\x8e;\xeaq\xf8ozU:\x90\x9b\x9c\x8b\'B\xe4]9\xf2\xd3\x05x\xcbQ\x15>\x0f\x05<\x03_\xb6+D\xc1\xa9\x86\xa0\xc5\x05\xa4\xadJ\x02\xe2\x951\x0c\xb5D\x15RQ\xb2\xd9\x9b0@\xd0b\x08\xf7\n\xefH\t\xb5}E\x8d\x91\x1b\xf3\xb7\xcaB\xa1\xcd\'\xedE\x10\xa7\xfaf\x1b\xaa\xe0\xeehy\xda+q\xaf\xa7\xbc\xc9\xb8\x8eTA\xe7]\x0fo\xbfG\xb8\th`\xc6rx|{<\x0b\xb3\r,\x83\xf3\xe0\xcbz\x14\xef\xea\xfd\x95N\\\xde\x8f\x83Y\x8aVr=\xed\x1b\x92j6\xcd\x85\xe7\xa7\xd1L{Dv\xfe*\xa9\x9c\x84\x8d)\xd8\xae%\xbdx\xa5\xe1}\xbe\xc8\xc6p<E\xc9y\xdb\x17\xa0v\xfe\xf6vih\xf5\x95\xe8\x1c\x95g\x8e\xcf\xcd\x9e\xb6\xed\xa5O\xfca\x94{\x08\x1aV\xd0\x84\x8b\x87\x98[\xb2j\x0f\xd5\x15\x8b\xa1\x08\r\xb5\xb5\xee\x175\xd1k\xefro\x16\xd7\x0b\x00\xfaS)\x03\n\x8b\xc6^9\x03\xc8_\x0b\x86\xef\x0c\xd9\x07\xb6\x86\xfe8\\i\x08e\xf3\xa6\xa9\xc4\x05\xf4\x81\x9a\xd1\x96\xfe\xd7\xd3Ko\xc5G\xc7T\x94\x12\xc7G\xce\x1e\x03\xa2F:dZsDW\xdcP\x89v\x8f\xca\xe9\x15\x8ac\xd6'

asn1 = asn1crypto.x509.Certificate.load(x509)
asn1.native
```

**Expected behavior**
Raise ValueError.

**Version:**
 - OS: Ubuntu 19.04
 - Python: 3.6.8
 - asn1crypto==0.24.0

**Additional context**
This was the original PEM and how I got it to x509:
```
import OpenSSL.crypto

PEM = '-----BEGIN CERTIFICATE-----\nMIIE8DCCAtigAwIBAgIQGzYKLRPFPl1J7bPO0csSfjANBgkqhkiG9w0BAQsFADAS\nMRAwDgYDVQQKEwdBY21lIENvMB4XDTE5MDgyMDE1NTAxOFoXDTIwMDgxOTE1NTAx\nOFowEjEQMA4GA1UEChMHQWNtZSBDbzCCAiIwDQYJKoZIhvcNAQEBBQADggIPADCC\nAgoCggIBAJwb4uXonxji9xMhqqgJZVO7uwtvq4rC9enNg05XJpPjweapIG9QG/kl\nJvGSWpPQM5O+Dm0E2rfq+tEnNwHIPCr+7H1Kk6B8c1N+js44ONT4nUh0msdyRq8i\n8QieEaCfvG8WTcmc/a9oIi7jbDD07EOOMCY+wd4JhQyF1xkCF7TdlT6UUIEQ2gxM\nazmTx3UDGrkIRKPnhdiiaFP5rkNtB61JR/9TBStNfD6oKFTGua3rF/wKGB2QfoT4\nYY/LSuS72TRJp2gGFjXbRiCKK55GRW0Gykrn0QuPylkOtSO8CHdAsr4tT9ENxJqW\nH3zNYU/kjs2NymxB9q3osWu3gN2dwl2i2t9cN2VwsH8yxo8QSwZbELljg8f/dCSO\nwTbYgjBkj8oYIdROG1l9H0lp+GYtDfy2y+yVrsAXgObM4KsfdkIsZ7kQ+SJxym22\n3jNevshvP37RioJiWuhStwwA+h0jOWGs3i25ua+n+FUtZD9WsqWH0uwtgLkm95r4\nSoegU3pzFqU4MhA/p5fFQ4WRF0vq3iV9cooZ7a+HfU3iqHS+J/WlJqiL14Xlpaje\nIo+lxZYcC1tfvHER13EzpLTvDKlH60V37Fh01yn4AaJlB8jpiBoUX2R28V1+3S2m\nFBmKXTpE8Kn/RTLeLxUFIUzt9qT7Ko0ae3CDWedLju0l9fNDiKjjAgMBAAGjQjBA\nMA4GA1UdDwEB/wQEAwIFoDATBgNVHSUEDDAKBggrBgEFBQcDATAMBgNVHRMBAf8E\nAjAAMAsGA1UdEQQEMAKHADANBgkqhkiG9w0BAQsFAAOCAgEAEyDgRKmGhHFP1G8K\nB0kz3cjTP8gmAbeYkN/8ZZ1ty2+tONmA8RW0sjhHrCWB2/n+Mgkr0uu/5k3LJ8gk\nkvHkOHvxre72MG7VoZA0mIR0pX1uNP3veohgmi+iqZzbMfkaWBMBVEAi3H+TSsuL\nNp2FeSA/bics/ddkhiXN4L11Gw32dfwp1a8xUxIRDZjTuTeSGSAGUE213ksEmq94\n9lwS20vZ3xl2Y8+cSTmRuqaiHGB6QMa8UdPGeqDyT/6on6Ia4DOzHyYrU4qzmD2O\nO+px+G96VTqQm5yLJ0LkXTny0wV4y1EVPg8FPANftitEwamGoMUFpK1KAuKVMQy1\nRBVSUbLZmzBA0GII9wrvSAm1fUWNkRvzt8pCoc0n7UUQp/pmG6rg7mh52itxr6e8\nybiOVEHnXQ9vv0e4CWhgxnJ4fHs8C7MNLIPz4Mt6FO/q/ZVOXN6Pg1mKVnI97RuS\najbNheen0Ux7RHb+KqmchI0p2K4lvXil4X2+yMZwPEXJedsXoHb+9nZpaPWV6ByV\nZ47PzZ627aVP/GGUewgaVtCEi4eYW7JqD9UVi6EIDbW17hc10Wvvcm8W1wsA+lMp\nAwqLxl45A8hfC4bvDNkHtob+OFxpCGXzpqnEBfSBmtGW/tfTS2/FR8dUlBLHR84e\nA6JGOmRac0RX3FCJdo/K6RWKY9Y=\n-----END CERTIFICATE-----\n'

x509 = OpenSSL.crypto.load_certificate(OpenSSL.crypto.FILETYPE_PEM, PEM)```

**Comment by wbond** [comment]
Thanks for the bug report. I'll check this out soon.

**Comment by joernheissler** [comment]
Hello,
your certificate is wrong. How did you generate it?

If you dump it with `openssl x509 -text -noout`, you get:
```
            X509v3 Subject Alternative Name: 
                IP Address:<invalid>
```

Smallest code to reproduce this with asn1crypto:
```python
from asn1crypto.x509 import IPAddress
IPAddress.load(b'\x04\x00').native
```

Or using an actuall snippet of your cert:
```python
from asn1crypto.x509 import GeneralNames
GeneralNames.load(b'\x30\x02\x87\x00').native
```

**Comment by steven2308** [comment]
Yes, I assumed it was wrong. I found it on a website (I also found a few other cases), the corresponding PEM is on the report too.
However an UnboundLocalError is not really the problem, so I simply suggest to raise ValueError instead.

**Comment by wbond** [comment]
The UnboundLocalError is fixed by 011ab06123a33e914262617d6a784c1d95d53f0f.

The idea of adding a ValueError in this situation would probably be part of a larger ticket to discuss data validation, which currently asn1crypto doesn't really do much with.
